### PR TITLE
[WM-2445] Add github_method_details table and branch_or_tag column

### PR DESCRIPTION
**Pull request by kpierre13**
image

### CODE REVIEW

File: service/src/main/resources/changelog/changesets/20240126_github_method_details_batch_or_tag.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: organization
+                  type: varchar(100)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: path
+                  type: varchar(1000)
```
Comment (salonishah11) [inline service/src/main/resources/changelog/changesets/20240126_github_method_details_batch_or_tag.yaml:21]: TOL: Should this and above `varchar` also be `text`? Is there a benefit of `varchar` over `text`?

**Comment by cjllanwarne** [comment]
I thought we were only going to store the `branch_name` column for now, and leave "branch or tag" to a later workstream around method versioning/saving configurations

**Comment by kpierre13** [comment]
> I thought we were only going to store the `branch_name` column for now, and leave "branch or tag" to a later workstream around method versioning/saving configurations

Are these two different columns? I named it branch or tag to more accurately represent what it'll be used for, since when we backfill this data, there's no telling whether or not users have been using branches or tags.

**Comment by cjllanwarne** [comment]
>Are these two different columns? I named it branch or tag to more accurately represent what it'll be used for, since when we backfill this data, there's no telling whether or not users have been using branches or tags.

Maybe `branch_or_tag_name`? To me `branch_or_tag` would be a boolean or an enum with two possible values: `branch` and `tag`

**Comment by kpierre13** [comment]
> Maybe `branch_or_tag_name`? To me `branch_or_tag` would be a boolean or an enum with two possible values: `branch` and `tag`

Currently in this PR, the column name is `branch_or_tag_name`. Is there another place I missed that should be renamed?

**Comment by cjllanwarne** [comment]
Oh, no, that sounds fine then... I think I was just going off the ticket/pr title

File: service/src/main/resources/changelog/changesets/20231102_set_table_role.yaml [hunk 2]
```diff
@@ -20,3 +20,4 @@ databaseChangeLog:
           ALTER TABLE method_version OWNER TO ${dbRole};
           ALTER TABLE run OWNER TO ${dbRole};
           ALTER TABLE run_set OWNER TO ${dbRole};
+          ALTER TABLE github_method_details OWNER TO ${dbRole};
```
Comment (cjllanwarne) [inline service/src/main/resources/changelog/changesets/20231102_set_table_role.yaml:23]: Thanks for remembering this!

File: service/src/main/resources/changelog/changelog.yaml [hunk 3]
```diff
@@ -89,6 +89,9 @@ databaseChangeLog:
   - include:
       file: changesets/20231102_set_table_role.yaml
       relativeToChangelogFile: true
```
Comment (cjllanwarne) [inline service/src/main/resources/changelog/changelog.yaml:91]: The set_table_role changeset is the actual changeset that needs to be at the end. It looks like _someone_ (oops!) forgot to honor this the last time the file was updated. So the order should go:
- add_workspace_id
- github_method_details
- set_table_role

And I wonder whether we can make the reminder more easy to notice somehow 🤔

**Review by cjllanwarne — CHANGES_REQUESTED** [review]
Requesting changes but they're not your fault

File: service/src/main/resources/changelog/changelog.yaml (hunk 3 above)
Comment (kpierre13) [reply service/src/main/resources/changelog/changelog.yaml:91]: I added the name of the change set in the reminder to maybe make it easier to notice?

**Review by cjllanwarne — APPROVED** [review]

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**